apisix/plugins/real-ip.lua

```diff
@@ -67,6 +73,19 @@ function _M.check_schema(conf)
 end
 
 
+local function addr_match(conf, addr)
+    if not conf.trusted_addresses then
+        return false
+    end
+
+    if not matcher then
```

Comment:
We should not use global matcher. Otherwise different configuration will use the same matcher.

Reply:
But I defined it as a local variable above.

Reply:
Different requests with different configurations will share this as it is defined as a module variable.

Reply:
I see, can I use lru cache to solve it? Or do you have any suggestions?

Reply:
Use lru cache instead, refer to the implementation of ip-restriction